content/en/hosting/4.x/upgrade-troubleshooting.md

```diff
@@ -0,0 +1,153 @@
+
+{{% pageinfo %}}
+All tips apply to both [Docker]({{< relref "hosting/4.x/production/docker" >}}) and [Kubernetes]({{< relref "hosting/4.x/production/kubernetes" >}}) based deployments unless otherwise specified.
+
+All upgrades are expected to succeed without issue.  Do not attempt any fixes unless you actively have a problem upgrading.
+{{% /pageinfo %}}
+
+## Considerations
+
+When troubleshooting, consider making sure there are:
+
+* Backups exist and restores have been tested 
+* Extra disk space is availabe (up to 5x!)
+* The upgrade has been tested on a development instance with production data
+
+## A go-to fix: restart
+
+A safe fix for any upgrade getting stuck is to restart all services.  Any views that were being re-indexed will be picked up where they left off without losing any work.  This should be your first step when trouble shooting a stuck upgrade. 
+
+If you're able to, after a restart go back into the admin web GUI and try to upgrade again.  Consider trying this at least twice.
+
+## CHT 4.0.x - 4.3.x: CouchDB Crashes
+
+**[Issue #9286](https://github.com/medic/cht-core/issues/9286)**:  Starting an upgrade that involves view indexing can cause CouchDB to crash on large databases (>30m docs).  The upgrade will fail and you will see the logs below when you have this issue.
+
+HAProxy:
+
+```shell
+<150>Jul 26 20:57:39 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+<150>Jul 26 20:57:40 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+<150>Jul 26 20:57:41 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+```
+
+CouchDB
+
+```shell
+[notice] 2024-07-26T20:52:45.229027Z couchdb@127.0.0.1 <0.10998.4> dca1387a05 haproxy:5984 172.22.0.4 medic GET /_active_tasks 200 ok 1
+[notice] 2024-07-26T20:52:45.234397Z couchdb@127.0.0.1 <0.10986.4> 2715cd1e47 haproxy:5984 172.22.0.4 medic GET /medic-logs/_all_docs?descending=true&include_docs=true&startkey=%22upgrade_log%3A1722027165223%3A%22&limit=1 200 ok 6
+[notice] 2024-07-26T20:52:45.468469Z couchdb@127.0.0.1 <0.11029.4> 3bf85c6071 haproxy:5984 172.22.0.4 medic GET /_active_tasks 200 ok 1
+```
+
+**Fix:**
+1. Check that all the indexes are warmed by loading them one by one in fauxton.
+2. Restart all services, **retry** upgrade from Admin GUI - do not cancel and upgrade.
+
+## CHT 4.0.0 - 4.2.2: view indexing can become stuck after indexing is finished
+
+**[Issue #9617](https://github.com/medic/cht-core/issues/9617):** Starting an upgrade that involves view indexing can become stuck after indexing is finished
+
+Upgrade process stalls while trying to index staged views:
+
+![CHT Core admin UI showing upgrade progress bar stalled at 4% ](stalled-upgrade.png)
+
+**Fix:**
+
+Unfortunately, the workaround is manual and very technical and involves:
```

Comment:
> the workaround is manual and very technical and involves:

I believe this remained as a statement from the time when we instructed to manually edit upgrade logs. Is this "very technical"?

Reply:
fixed in fc5f2faf - thanks!